File: Visual Studio/C#/Eruru.MVVM/Shared/MVVMRelayCommand.cs
using System;

namespace Eruru.MVVM {

	public class MVVMRelayCommand {

		readonly Predicate<object> _CanExecute;
		readonly Action<object> _Execute;

		public MVVMRelayCommand (Action<object> execute, Predicate<object> canExecute) {
			_Execute = execute;
			_CanExecute = canExecute;
		}
		public MVVMRelayCommand (Action<object> execute) {
			_Execute = execute;
		}

		public bool CanExecute (object parameter) {
			return _CanExecute == null ? true : _CanExecute (parameter);
		}

		public void Execute (object parameter) {
			_Execute?.Invoke (parameter);
		}

	}

}

File: WebStorm/JavaScript/eruru.mvvm/MVVMRelayCommand.js
function MVVMRelayCommand  (execute, canExecute) {

	this._execute = execute;
	this._canExecute = canExecute;

	this.canExecute = function (parameter) {
		return this._canExecute == null || this._canExecute (parameter);
	};

	this.execute = function (parameter) {
		if (this._execute != null) {
			this._execute (parameter);
		}
	};

}